silkworm/sync/internals/chain_elements.hpp

```diff
@@ -261,8 +261,8 @@ struct Segment
     using Slice = std::span<const HeaderList::Header_Ref>;  // a Segment slice
 
     [[nodiscard]] Slice slice(size_t start, size_t end) const {
-        return Slice(*this).subspan(start, end - start);
-    }  // with c++20 it can be implemented as: return Slice(begin() + start, begin() + end);
+        return Slice(&(*this)[start], &(*this)[end]);
```

Comment:
I'd suggest using `return Slice(begin() + start, begin() + end);` again for readability

Reply:
For some reason the stdlib that the clang-tidy build uses doesn't contain any span constructors other than `slice(ptr, count)` and `slice(ptr1, ptr2)`, so passing an iterator like `begin() + start` fails compilation, but I can try using `data() + start`.